### PR TITLE
81 Dynamics: Installer Hashes

### LINKED ISSUE #81: Dynamics: Installer Hashes
Client should:

- Attempt to pull latest hashes on startup
- Attempt to refresh hashes every `X` hours after startup
- Error Handle offline scenario
- Error Handle non-`200` server responses
- Error Handle malformed `200` server responses
- Revert to local `yaml` file on failure

Server & Endpoint for testing: https://hash-server-7be6da1b0395.herokuapp.com/installer_hashes

```
$ curl https://hash-server-7be6da1b0395.herokuapp.com/installer_hashes
---
6c28ed9f08c2bfdd1a9ccb5da26ab27549d3d95ac791e013c3316264a0b7a2eb: BlockstreamGreen_Windows_x86_64.zip
1b8589bc997016087ba27ef28f2026626834c91af525ca88bd3841bafcbe4c06: BlockstreamGreen_MacOS_x86_64.zip
.
.
.
cdd03c92a299d7251fd2e272cec9b0f10c233ca5e0be28af14fcd6ffd9e2c838: whirlpool-gui-0.10.4.AppImage
```

### PR DESCRIPTION
**Pull request by kylemccullen**
Closes #81

### CODE REVIEW

File: Branta/Automation/Installer.cs [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
+    private Dictionary<string, string> _hashes;
+
+    public Installer(NotifyIcon notifyIcon, ResourceDictionary resourceDictionary) : base(notifyIcon, null, 60)
     {
         _notifyIcon = notifyIcon;
         _resourceDictionary = resourceDictionary;
+
+        _brantaHashClient = new HttpClient()
+        {
+            BaseAddress = new Uri("https://hash-server-7be6da1b0395.herokuapp.com")
```
Comment (kylemccullen) [inline Branta/Automation/Installer.cs:24]: The server address should be an AppSetting. Is this address subject to change in the future that would require a user to update to the latest version of Branta?

File: Branta/Automation/Installer.cs [hunk 2]
```diff
@@ -36,4 +44,29 @@ public void ProcessFiles(string[] files)
...
     }
+
+    public override void Run()
+    {
+        Task.Run(LoadInstallerHashesAsync);
+    }
+
+    private async Task LoadInstallerHashesAsync()
+    {
+        _hashes = await FetchInstallerHashes() ?? YamlLoader.LoadInstallerHashes();
```
Comment (kylemccullen) [inline Branta/Automation/Installer.cs:55]: If the server responds with a valid file, we should replace the local yaml file with the up-to-date server version.

Comment (keith-gardner) [reply Branta/Automation/Installer.cs:55]: Yes, we can either write the yaml to disk, or just cache the contents. Writing to disk is better

File: Branta/Automation/Installer.cs (hunk 1 above)
Comment (keith-gardner) [reply Branta/Automation/Installer.cs:24]: 👍 

We'll dial in the URL before release. Once we release, there should be no reason to change the URL

If the URL is ever not responding, client needs to fail safely and use local `yaml`

**Review by keith-gardner — APPROVED** [review]

**Comment by keith-gardner** [comment]
DNS is still propagating, `https://api.branta.pro/installer_hashes` should resolve everywhere by this evening

**Review by keith-gardner — APPROVED** [review]

File: Branta/Automation/Installer.cs (hunk 2 above)
Comment (kylemccullen) [reply Branta/Automation/Installer.cs]: Will be implemented in #88

**Pull request merged**